### PR TITLE
Fix +2 mace breaking filters

### PR DESCRIPTION
**Pull request by dekkerglen**
The problem: using filters when +2 mace is in the available pool causes the parser to crash.

The cause: when replacing the name in the oracle text with a placeholder, we treat the name as a regular expression, where `+` has special meaning

The solution: Treat any `+`'s in card names as string literals

### CODE REVIEW

**Review by ruler501 — CHANGES_REQUESTED** [review]
This should be a global replacement in the string for future proofing.

**Comment by dekkerglen** [comment]
No, there are many places the name is used not as a regex

**Comment by ruler501** [comment]
You misunderstood. You should replace all instances of + not just the first.

File: src/filtering/FuncOperations.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const nameStringOperation = (op, value) => {
   const strOp = stringOperation(op, value);
   return (fieldValue, card) => {
     let expandedValue = fieldValue
-      .replace(new RegExp(card.details.name, 'g'), NAME_PLACEHOLDER)
+      .replace(new RegExp(card.details.name.replace('+', '\\+'), 'g'), NAME_PLACEHOLDER)
```
Comment (lunakv) [inline src/filtering/FuncOperations.js:70]:
```suggestion
      .replace(new RegExp(card.details.name.replace(/+/g, '\\+'), 'g'), NAME_PLACEHOLDER)
```

Comment (lunakv) [reply src/filtering/FuncOperations.js:70]: by the way, this all is the result of Node not having a proper `replaceAll` function until v15, which is just dumb.

**Pull request merged**